monai/inferers/inferer.py

```diff
@@ -190,3 +192,53 @@ def __call__(
             *args,
             **kwargs,
         )
+
+
+class SaliencyInferer(Inferer):
+    """
+    SaliencyInferer is inference with activation maps.
+
+    Args:
+        cam_name: expected CAM method name, hould be: "CAM", "GradCAM" or "GradCAMpp".
+        target_layers: name of the model layer to generate the feature map.
+        class_idx: index of the class to be visualized. if None, default to argmax(logits).
+        args: other optional args to be passed to the `__init__` of cam.
+        kwargs: other optional keyword args to be passed to `__init__` of cam.
+
+    """
+
+    def __init__(self, cam_name: str, target_layers: str, class_idx: Optional[int] = None, *args, **kwargs) -> None:
+        Inferer.__init__(self)
+        if cam_name.lower() not in ("cam", "gradcam", "gradcampp"):
+            raise ValueError("cam_name should be: 'CAM', 'GradCAM' or 'GradCAMpp'.")
+        self.cam_name = cam_name.lower()
+        self.target_layers = target_layers
+        self.class_idx = class_idx
+        self.args = args
+        self.kwargs = kwargs
+
+    def __call__(
+        self,
+        inputs: torch.Tensor,
+        network: nn.Module,
+        *args: Any,
+        **kwargs: Any,
+    ):
+        """Unified callable function API of Inferers.
+
+        Args:
+            inputs: model input data for inference.
+            network: target model to execute inference.
+                supports callables such as ``lambda x: my_torch_model(x, additional_config)``
+            args: other optional args to be passed to the `__call__` of cam.
+            kwargs: other optional keyword args to be passed to `__call__` of cam.
+
+        """
+        if self.cam_name == "cam":
+            cam = CAM(network, self.target_layers, *self.args, **self.kwargs)
```

Comment:
instead of passing by name, couldn't they construct the object themselves and pass that as an argument? Would save you storing `args` and `kwargs`.

Reply:
Hi @rijobro ,

Thanks for your suggestion, I thought about it, but there are 2 minor issues with it:
1. Our Clara and other projects are based on MONAI API and using JSON config, it's hard to create and pass "object" as arg in JSON.
2. To create `CAM` or `GradCAM` object, we need the `network` as parameter, which is provided at runtime with the "__call__" of inferer.

Anyway, maybe we can refactor the inferer design to support more flexible features later.

Thanks.